### DISCUSSION TITLE
What is the "Day Time" Logic for Theme "Auto"?

### DISCUSSION THREAD

**Discussion by wesleyboar-fka-iosulfur** [gsantner/markor#1433] [Q&A] [answered]
### General Information

* __App version__: v2.8.0, Flavor: Gpjay (release)
* __System__: Pixel 5 with Android 11

### Question

* How does the app determine “day time” for supporting Theme “Auto” logic?

### Experience

The sun had long since risen, but app still determined Auto should result in Dark mode. Later, the app had begun to determine Auto should result in Light mode.

#### How Theme Does _Not_ Work

The Theme “Auto” does [not](https://github.com/gsantner/markor/issues/1318#issue-860590176) trigger based on [recent UI guidelines](https://github.com/gsantner/markor/issues/1318#issuecomment-821946586), because the “[current implementation works in the same way for any OS version](https://github.com/gsantner/markor/issues/1318#issuecomment-821967283)” ([for simplicity](https://github.com/gsantner/markor/issues/1121#issuecomment-724639932)).

#### How Theme Works (Simplified)

The ["auto mode is [determined] by day time, nothing else"](https://github.com/gsantner/markor/issues/1318#issuecomment-821944480).

### Related Issues
* https://github.com/gsantner/markor/issues/1318
* https://github.com/gsantner/markor/issues/1121
* https://github.com/gsantner/markor/issues/1009

**Comment by gsantner** [comment, marked as answer]
The logic is based on hours -> <https://github.com/gsantner/markor/blob/master/app/src/main/java/net/gsantner/markor/util/AppSettings.java#L69>.

Why? This Android own feature is pretty new, and many devices dont support it yet. It's an effort to support both "the new" and have a fallback that just works everywhere. So instead, we just have something that "just works everywhere".

If somebody wants to put time into this, and provide a COMPLETE solution for current and old devices, please do so and make a pull request. I won't do this.

**Comment by wesleyboar-fka-iosulfur** [comment]
9 am to 5 pm. Thank you.
